brainlit/algorithms/generate_fragments/spline_fxns.py

```diff
@@ -0,0 +1,296 @@
+import operator
+from scipy.interpolate import BSpline, splev
+import numpy as np
+from typing import Iterable, Union, Tuple
+from brainlit.utils.util import (
+    check_type,
+    check_size,
+    check_precomputed,
+    check_iterable_type,
+    check_iterable_nonnegative,
+)
+
+
+def speed(
+    x: np.ndarray,
+    t: np.ndarray,
+    c: np.ndarray,
+    k: np.integer,
+    aux_outputs: bool = False,
+) -> np.ndarray:
+    """Compute the speed of a B-Spline.
+
+    The speed is the norm of the first derivative of the B-Spline
+
+    Arguments:
+        x: A 1xL array of parameter values where to evaluate the curve
+        t: A 1xm array representing the knots of the B-spline
+        c: A dxn array representing the control points of the B-spline
```

Comment:
are you sure one of the dimensions is not m, in order to match the length of "t"?

Reply:
if so, please throw errors if they don't match

Reply:
I am not sure:

- `x` does not have to to match the length of `t`
- `m = n + p +1` for the degree of the spline to match [[1](https://mathworld.wolfram.com/B-Spline.html)], and this is checked in [`scipy.interpolate.BSpline`](https://github.com/scipy/scipy/blob/79e64c140a518bc85d9105718301509f43e51444/scipy/interpolate/_bsplines.py#L171-L213) already.

Does this make sense to you too?